docs: add model coverage tables (#226)
NVIDIA-NeMo/Automodel: docs/model_coverage/vlm.md

# Vision Language Models with NeMo AutoModel

## Introduction

Vision Language Models (VLMs) are advanced models that integrate vision and language processing capabilities. They are trained on extensive datasets containing both interleaved images and text data, allowing them to generate text descriptions of images and answer questions related to images.

NeMo AutoModel LLM APIs can be easily extended to support VLM tasks. While most of the training setup is the same, some additional steps are required to prepare the data and model for VLM training.

## Run LLMs with NeMo Automodel

> To run LLMs with NeMo AutoModel, please use at least version `25.07` of the NeMo container.
> If the model you want to finetune is available on a newer version of transformers, you may need
> to upgrade to the latest NeMo Automodel with:

Reviewer: ```suggestion
To run LLMs with NeMo AutoModel, use NeMo container version `25.07` or later. If the model you want to fine-tune requires a newer version of Transformers, you may need to upgrade to the latest NeMo AutoModel using:
```

## Dataset Examples

> [!TIP]
> In these guides, we use the `quintend/rdr-items` and `naver-clova-ix/cord-v2` datasets for demonstation purposes, but you can specify your own data as needed.

### rdr items dataset
The rdr items dataset [`quintend/rdr-items`](https://huggingface.co/datasets/quintend/rdr-items) is a small dataset containing 48 images with descriptions. This dataset serves as an example of how to prepare image-text data for VLM fine-tuning. For complete instructions on dataset preprocessing and the collate functions used, see the [Gemma Fine-Tuning Guide](guides/omni/gemma3-3n.md).

### cord-v2 dataset
The cord-v2 dataset [`naver-clova-ix/cord-v2`](https://huggingface.co/naver-clova-ix/cord-v2) contains receipts with descriptions in JSON format. This demonstrates handling structured data in VLMs. The [Gemma Fine-Tuning Guide](guides/omni/gemma3-3n.md) provides detailed examples of custom preprocessing and collate functions for similar datasets.

> ## Training VLM models

Reviewer: ```suggestion
## Train VLM Models
```

All supported models can be fine-tuned using either full SFT or PEFT approaches. The [Gemma Fine-Tuning Guide](guides/omni/gemma3-3n.md) provides complete instructions for:

> * Configuring YAML-based training

Reviewer: ```suggestion
* Configuring YAML-based training.
```

> * Running single-GPU and multi-GPU training

Reviewer: ```suggestion
* Running single-GPU and multi-GPU training.
```

> * Setting up parameter-efficient fine-tuning with LoRA

Reviewer: ```suggestion
* Setting up PEFT with LoRA.
```

> * Model checkpointing and W&B integration

Reviewer: ```suggestion
* Model checkpointing and W&B integration.
```
